diff -urNp -x '*.orig' grpc-1.32.0.org/setup.py grpc-1.32.0/setup.py
--- grpc-1.32.0.org/setup.py	2021-03-04 23:29:29.994075500 +0100
+++ grpc-1.32.0/setup.py	2021-03-04 23:29:30.220742787 +0100
@@ -106,7 +106,7 @@ CLASSIFIERS = [
 ]
 
 BUILD_WITH_BORING_SSL_ASM = os.environ.get('GRPC_BUILD_WITH_BORING_SSL_ASM',
-                                           True)
+                                           False)
 
 # Environment variable to determine whether or not the Cython extension should
 # *use* Cython or use the generated C files. Note that this requires the C files
