diff -urNp -x '*.orig' grpc-1.32.0.org/setup.py grpc-1.32.0/setup.py
--- grpc-1.32.0.org/setup.py	2021-03-04 23:29:29.994075500 +0100
+++ grpc-1.32.0/setup.py	2021-03-04 23:29:30.220742787 +0100
@@ -106,7 +106,7 @@ CLASSIFIERS = [
 
 
 BUILD_WITH_BORING_SSL_ASM = _env_bool_value('GRPC_BUILD_WITH_BORING_SSL_ASM',
-                                            'True')
+                                            'False')
 
 # Export this environment variable to override the platform variant that will
 # be chosen for boringssl assembly optimizations. This option is useful when
