--- grpc-1.78.1/setup.py.orig	2026-02-19 13:15:33.000000000 +0100
+++ grpc-1.78.1/setup.py	2026-03-01 08:12:06.044418753 +0100
@@ -548,14 +548,14 @@ def cython_extensions_and_necessity():
 
 CYTHON_EXTENSION_MODULES, need_cython = cython_extensions_and_necessity()
 
-INSTALL_REQUIRES = ("typing-extensions~=4.12",)
+INSTALL_REQUIRES = ("typing-extensions>=4.12",)
 
 EXTRAS_REQUIRES = {
     "protobuf": "grpcio-tools>={version}".format(version=grpc_version.VERSION),
 }
 
 SETUP_REQUIRES = (
-    INSTALL_REQUIRES + ("Sphinx~=1.8.1",) if ENABLE_DOCUMENTATION_BUILD else ()
+    INSTALL_REQUIRES + ("Sphinx>=1.8.1",) if ENABLE_DOCUMENTATION_BUILD else ()
 )
 
 try:
@@ -571,7 +571,7 @@ except ImportError:
         sys.stderr.write(
             "We could not find Cython. Setup may take 10-20 minutes.\n"
         )
-        SETUP_REQUIRES += ("cython==3.1.1",)
+        SETUP_REQUIRES += ("cython>=3.1.1",)
 
 COMMAND_CLASS = {
     "doc": commands.SphinxDocumentation,
